src/Microsoft.Extensions.FileSystemGlobbing/PatternMatchingResult.cs

```diff
@@ -15,13 +16,30 @@ public class PatternMatchingResult
         /// </summary>
         /// <param name="files">A collection of <see cref="FilePatternMatch" /></param>
         public PatternMatchingResult(IEnumerable<FilePatternMatch> files)
+            : this(files, hasMatches: files.Any())
         {
             Files = files;
         }
 
         /// <summary>
+        /// Initializes the result with a collection of <see cref="FilePatternMatch" />
+        /// </summary>
+        /// <param name="files">A collection of <see cref="FilePatternMatch" /></param>
+        /// <param name="hasMatches">A value that determines if <see cref="PatternMatchingResult"/> has any matches.</param>
+        public PatternMatchingResult(IEnumerable<FilePatternMatch> files, bool hasMatches)
+        {
+            Files = files;
+            HasMatches = hasMatches;
+        }
+
+        /// <summary>
         /// A collection of <see cref="FilePatternMatch" />
         /// </summary>
         public IEnumerable<FilePatternMatch> Files { get; set; }
+
+        /// <summary>
+        /// Gets a value that determines if this instance of <see cref="PatternMatchingResult"/> has any matches.
+        /// </summary>
+        public bool HasMatches { get; }
```

Comment:
In that case, we should ensure the symmetry of `HasChanges` and `Files.Any()` by removing the constructor that takes `bool hasMatches`. We could just initialize this value by calling Files.Any() once in the original ctor